packages/overlay/test/renderer.test.js

```diff
@@ -1,114 +1,147 @@
-        expect(model).to.eql(renderModel);
-      };
-    });
-
-    it('should clean the root on renderer changed', () => {
-      overlay.renderer = (root) => root.appendChild(rendererContent);
-      overlay.opened = true;
-      expect(overlay.textContent.trim()).to.equal('renderer-content');
-      overlay.renderer = (root) => expect(root.firstChild).to.be.null;
-    });
-
-    it('should not clean the root on model or owner changed', () => {
-      overlay.renderer = (root, owner, model) => {
-        if (owner !== undefined || model !== undefined) {
-          expect(root.firstChild).not.to.be.null;
-        }
-        root.appendChild(rendererContent);
-      };
-      const overlayOwner = {};
-      overlay.owner = overlayOwner;
-      overlay.model = renderModel;
-    });
-
-    it('should pass owner as this to the renderer', () => {
-      overlay.owner = {};
-      overlay.model = renderModel;
-      overlay.renderer = function (root, owner) {
-        expect(this).to.eql(owner);
-      };
-    });
-
-    it('should call renderer on model change', () => {
-      const spy = sinon.spy();
-      overlay.opened = true;
-      overlay.renderer = () => spy();
-      spy.resetHistory();
-      overlay.model = {};
-      expect(spy.calledOnce).to.be.true;
-    });
-
-    it('should call renderer on owner change', () => {
-      const spy = sinon.spy();
-      overlay.opened = true;
-      overlay.renderer = () => spy();
-      spy.resetHistory();
-      overlay.owner = {};
-      expect(spy.calledOnce).to.be.true;
-    });
-
-    it('should run renderers when requesting content update', () => {
-      overlay.renderer = sinon.spy();
-      overlay.requestContentUpdate();
-
-      expect(overlay.renderer.calledOnce).to.be.true;
-    });
-
-    it('should not render if overlay is not open', () => {
-      const spy = sinon.spy();
-      overlay.renderer = () => spy();
-      expect(spy.called).to.be.false;
-    });
-
-    it('should clear the content when removing the renderer', () => {
-      overlay.renderer = (root) => {
-        root.innerHTML = 'foo';
-      };
-      overlay.opened = true;
-
-      expect(overlay.textContent.trim()).to.equal('foo');
-
-      overlay.renderer = null;
-
-      expect(overlay.textContent.trim()).to.equal('');
-    });
+  afterEach(async () => {
+    overlay.opened = false;
+    await nextRender();
+  });
+
+  it('should use renderer when it is defined', async () => {
+    overlay.renderer = (root) => root.appendChild(content);
+    overlay.opened = true;
+    await nextRender();
+    expect(overlay.textContent.trim()).to.equal('renderer-content');
+  });
+
+  it('should receive empty root, model and owner when they are defined', async () => {
+    const overlayOwner = {};
+    const overlayModel = {};
+
+    overlay.owner = overlayOwner;
+    overlay.model = overlayModel;
+
+    const renderer = sinon.spy();
+
+    overlay.renderer = renderer;
+    overlay.opened = true;
```

Comment:
This test was a false positive: `renderer` was not invoked, because we didn't set `opened` to `true`.